datafusion/src/scalar.rs

```diff
@@ -100,6 +102,12 @@ impl PartialEq for ScalarValue {
         // any newly added enum variant will require editing this list
         // or else face a compile error
         match (self, other) {
+            (Decimal128(v1, p1, s1), Decimal128(v2, p2, s2)) => {
+                v1.eq(v2) && p1.eq(p2) && s1.eq(s2)
+                // TODO how to handle this case: decimal(123,10,1) with decimal(1230,10,2)
```

Comment:
`Int8(Some(100))` is not equal to `Int64(Some(100))` with this implementation either , so I think it is consistent with the rest of this comparison of `decimal(123,10,1) and `decimal(1230,10,2)` are different --

Reply:
yep.
If we want to compare two values with the different data types, you should convert them to the same data type and then compare them.